### PR TITLE
Implement tpm2 tests using swtpm device

### PR DESCRIPTION
**Pull request by rfan1**
Currently, we use IBM's Software TPM 2.0 to emulate TPM "device".
now,we have an alternative method to test tpm2 in qemu env which
uses swtpm device.

- Related ticket: https://progress.opensuse.org/issues/105732
- Needles: n/a
- Verification run: 
https://openqa.suse.de/tests/8157351            -SLE
https://openqa.opensuse.org/tests/2189468 -TW
- Regression test:
https://openqa.suse.de/tests/8168760           -SLE
https://openqa.opensuse.org/tests/2189459 -TW

### CODE REVIEW

File: tests/security/tpm2/tpm2_tools/tpm2_tools_auth.pm [hunk 1]
```diff
@@ -23,45 +23,45 @@ sub run {
     my $key_pub = "key.pub";
     my $key_name = "key.name";
     my $ket_ctx = "key.ctx";
+
+    my $tpm_suffix = '';
+    $tpm_suffix .= '-T tabrmd' if (get_var('QEMUTPM', 0) != 1 || get_var('QEMUTPM_VER', '') ne '2.0');
```
Comment (Amrysliu) [inline tests/security/tpm2/tpm2_tools/tpm2_tools_auth.pm:28]: `$tmp_suffix` is already an empty string in the above line. so I think just using `$tmp_suffix = '-T tabrmd'` should be ok.

File: tests/security/tpm2/tpm2_tools/tpm2_tools_encrypt.pm [hunk 2]
```diff
@@ -16,49 +16,49 @@ sub run {
     my $self = shift;
     $self->select_serial_terminal;
 
+    my $tpm_suffix = '';
+    $tpm_suffix .= '-T tabrmd' if (get_var('QEMUTPM', 0) != 1 || get_var('QEMUTPM_VER', '') ne '2.0');
```
Comment (Amrysliu) [inline tests/security/tpm2/tpm2_tools/tpm2_tools_encrypt.pm:20]: same comment as above

File: tests/security/tpm2/tpm2_tools/tpm2_tools_self_contain_tool.pm [hunk 3]
```diff
@@ -16,10 +16,13 @@ sub run {
     my $self = shift;
     $self->select_serial_terminal;
 
+    my $tpm_suffix = '';
+    $tpm_suffix .= '-T tabrmd' if (get_var('QEMUTPM', 0) != 1 || get_var('QEMUTPM_VER', '') ne '2.0');
```
Comment (Amrysliu) [inline tests/security/tpm2/tpm2_tools/tpm2_tools_self_contain_tool.pm:20]: same comment as above

File: tests/security/tpm2/tpm2_tools/tpm2_tools_sign_verify.pm [hunk 4]
```diff
@@ -16,6 +16,9 @@ sub run {
     my $self = shift;
     $self->select_serial_terminal;
 
+    my $tpm_suffix = '';
+    $tpm_suffix .= '-T tabrmd' if (get_var('QEMUTPM', 0) != 1 || get_var('QEMUTPM_VER', '') ne '2.0');
```
Comment (Amrysliu) [inline tests/security/tpm2/tpm2_tools/tpm2_tools_sign_verify.pm:20]: same

**Review by lilyeyes — APPROVED** [review]
lgtm

File: tests/security/tpm2/tpm2_tools/tpm2_tools_sign_verify.pm (hunk 4 above)
Comment (rfan1) [reply tests/security/tpm2/tpm2_tools/tpm2_tools_sign_verify.pm]: Thanks,done

**Review by jouyingbin — APPROVED** [review]
LGTM

**Review by ShawnHa0 — APPROVED** [review]
LGTM

**Review by Amrysliu — APPROVED** [review]

**Pull request merged**